docker/docker_image_src.go

```diff
@@ -116,12 +142,77 @@ func (s *dockerImageSource) GetBlob(digest string) (io.ReadCloser, int64, error)
 }
 
 func (s *dockerImageSource) GetSignatures() ([][]byte, error) {
-	return [][]byte{}, nil
+	if s.c.signatureBase == nil { // Skip dealing with the manifest digest if not necessary.
+		return [][]byte{}, nil
```

Comment:
you can do `return nil, nil` right? the only thing you can do from a slice is to range over it (and `len(nil) == 0)`) or access it w/o checking its length in which case everything breaks - so, I think it's safe to just return `nil`.